examples/material_gallery/lib/gallery/home.dart

```diff
@@ -50,13 +50,15 @@ class GalleryHome extends StatefulWidget {
 }
 
 class GalleryHomeState extends State<GalleryHome> {
-  final Key _scrollableKey = new UniqueKey();
+  final Key _homeKey = new ValueKey("Gallery Home");
+  final Key _listKey = new ValueKey("Gallery List");
```

Comment:
It should ValueKey<String>, I'll fix that. And I have a confession: the code inexplicably fails if I used UniqueKey here. I noticed because I was debugging with ValueKey and then reverted to UniqueKey and then things went awry.

I will try and distill a proper test case as a followup to this change.